File: cardstackview/src/main/java/com/yuyakaido/android/cardstackview/Direction.java
package com.yuyakaido.android.cardstackview;

import java.util.Arrays;
import java.util.List;

public enum Direction {
    Left,
    Right,
    Top,
    Bottom;

    public static final List<Direction> HORIZONTAL = Arrays.asList(Direction.Left, Direction.Right);
    public static final List<Direction> VERTICAL = Arrays.asList(Direction.Top, Direction.Bottom);
    public static final List<Direction> FREEDOM = Arrays.asList(Direction.values());
}

File: cardstackviewcompose/src/main/java/com/yuyakaido/android/cardstackview/compose/Direction.kt
package com.yuyakaido.android.cardstackview.compose

sealed interface Direction {
    val isDrag: Boolean

    data class Top(override val isDrag: Boolean) : Direction
    data class Bottom(override val isDrag: Boolean) : Direction
    data class Left(override val isDrag: Boolean) : Direction
    data class Right(override val isDrag: Boolean) : Direction
}

